packages/siwx/exports/react.ts

```diff
@@ -0,0 +1,10 @@
+import { useMemo } from 'react'
+
+import { OptionsController, type SIWXConfig } from '@reown/appkit-controllers'
+
+export function useAppKitSIWX<Config extends SIWXConfig = SIWXConfig>() {
+  return useMemo(
+    () => OptionsController.state.siwx as Config | undefined,
+    [OptionsController.state.siwx]
```

Comment:
I guess to be able to make recalculated every tiem siwx object changed, you need to listen changes to it, when do that way, it'll calculated once since the `OptionsController.state.siwx` is not a dynamic value.

```suggestion
export function useAppKitSIWX<Config extends SIWXConfig = SIWXConfig>() {
  const siwx = useSnapshot(OptionsController.state.siwx)
  return useMemo(
    () => OptionsController.state.siwx as Config | undefined,
    [siwx]
  )
}
```

Reply:
https://github.com/reown-com/appkit/pull/4087/commits/8e5ee3ad25755e43b114f6c6e0696add333be3ed